modules/ServerAPI/src/server/python/acas_ldclient/acasldclient.py

```diff
@@ -0,0 +1,189 @@
+        help='LD client user',
+        dest='username',
+        required=True
+    )
+    parser.add_argument(
+        '--password',
+        '-p',
+        help='LD client password',
+        dest='password',
+        required=True
+    )
+    parser.add_argument(
+        '--method',
+        '-m',
+        help='Method to call',
+        dest='method',
+        required=True
+    )
+    parser.add_argument(
+        '--args',
+        '-a',
+        nargs='*',
+        help='arguments to the method',
+        dest='args',
+        required=False
+    )
+    return parser
+
+def list_groups(client, test):
+    print test
+    return client.list_groups()
+
+def auth_check(client, username, password):
+    auth_return = client.auth_check(username = username, password = password)
+    return auth_return
+
+def ld_group_to_acas_role(group):
+    # u'LDAP_Auto_ROLE_SEURAT-USERS'
+    # u'Project_ProjectX_Administrator'
+    split = group["name"].split('_')
+    ls_type = split[0]
+    ls_kind = split[1]
+    role_name = "_".join(split[2:len(split)])
+    acas_role = {
+        "id": group["id"],
+        "roleEntry": {
+            'id': group["id"],
+            'lsType': ls_type,
+            'lsKind': ls_kind,
+            'lsTypeAndKind': ls_type + "_" + ls_kind,
+            'roleDescription': "Original group fetched from Live Design: "+ group["name"],
+            'roleName': role_name,
+            'version': 0
+        }
+    }
+    return acas_role
+
+def ld_user_to_acas_user_code_table(ld_user):
+    acas_user = {
+        "code":ld_user["username"],
+        "id":ld_user["id"],
+        "ignored":False,
+        "name":ld_user["username"]
+    }
+    return acas_user
+
+def get_users(client, ls_type = None, ls_kind = None, role_name = None):
+    ld_users = client.list_users()
+    if ls_type == None and ls_kind == None and role_name == None:
+        acas_users = map(ld_user_to_acas_user_code_table, ld_users)
+    else:
+        groups = client.list_groups()
+        roles = map(ld_group_to_acas_role, groups)
+        role = [r for r in roles if r["roleEntry"]["lsType"] == ls_type and r["roleEntry"]["lsKind"] == ls_kind and r["roleEntry"]["roleName"] == role_name]
+        if len(role) == 0:
+            return []
+        users = []
+        memberships = client.list_memberships()
+        userDict = {}
+        for u in ld_users:
+            userDict[u["id"]] = u
+        for m in memberships:
+            if m["group_id"] == role[0]["id"]:
+                users.append(userDict[m["user_id"]])
+        acas_users = map(ld_user_to_acas_user_code_table, users) 
+    return acas_users
+
+def ld_user_to_acas_user(ld_user, roles):
+    acas_user = {
+        'id': ld_user["id"],
+        'username': ld_user["username"],
+        'email': ld_user["username"],
+        'firstName': ld_user["username"],
+        'lastName': ld_user["username"],
+        'roles': roles
+    }
+    return acas_user
+
+def get_users_roles(client, users):
```

Comment:
Nit: This might make more sense if you defined a python class that had "client" as an instance attribute (i.e. "self.client") rather than having to pass it into many functions as an argument. Just food for thought, not a required change.

Reply:
I will likely need to make more changes to this client so I will make this change then